src/hooks/useTransactionStatus.ts

```diff
@@ -1,22 +1,33 @@
 import { useAppSelector } from '@/store'
 import { PendingStatus, selectPendingTxById } from '@/store/pendingTxsSlice'
+import { isSignableBy } from '@/utils/transaction-guards'
 import { TransactionSummary, TransactionStatus } from '@gnosis.pm/safe-react-gateway-sdk'
+import { useMemo } from 'react'
+import useWallet from './wallets/useWallet'
 
 type TxLocalStatus = TransactionStatus | PendingStatus
 
-const STATUS_LABELS: Record<TxLocalStatus, string> = {
-  [TransactionStatus.AWAITING_CONFIRMATIONS]: 'Awaiting confirmations',
-  [TransactionStatus.AWAITING_EXECUTION]: 'Awaiting execution',
-  [TransactionStatus.CANCELLED]: 'Cancelled',
-  [TransactionStatus.FAILED]: 'Failed',
-  [TransactionStatus.SUCCESS]: 'Success',
-  [PendingStatus.SUBMITTING]: 'Submitting',
-  [PendingStatus.MINING]: 'Mining',
-  [PendingStatus.INDEXING]: 'Indexing',
-}
+const useTransactionStatus = (txSummary: TransactionSummary): string => {
+  const { txStatus, id } = txSummary
 
-const useTransactionStatus = ({ txStatus, id }: TransactionSummary): string => {
+  const wallet = useWallet()
   const pendingTx = useAppSelector((state) => selectPendingTxById(state, id))
+
+  const STATUS_LABELS: Record<TxLocalStatus, string> = useMemo(
+    () => ({
+      [TransactionStatus.AWAITING_CONFIRMATIONS]:
+        wallet?.address && isSignableBy(txSummary, wallet.address) ? 'Needs your confirmation' : 'Needs confirmations',
```

Comment:
I would keep the STATUS_LABELS static as they were and just add another dictionary like so:
```
const STATUS_LABELS: Record<TxLocalStatus, string> = { ... }

const WALLET_STATUS_LABELS: Record<TxLocalStatus, string> = {
  ...STATUS_LABELS,
  [TransactionStatus.AWAITING_CONFIRMATIONS]: 'Needs your confirmation'
}

const statuses = isSignableBy(txSummary, wallet.address) ? WALLET_STATUS_LABELS : STATUS_LABELS

return statuses[pendingTx?.status || txStatus] || '']
```